src/js/components/modals/AppModalComponent.jsx

```diff
@@ -166,7 +170,13 @@ var AppModalComponent = React.createClass({
   },
 
   handleAppConfigError: function (error) {
-    this.setState({error: error, appIsValid: false});
+    if (error != null ||
+        Object.keys(AppFormStore.validationErrorIndices).length) {
+      this.setState({
+        error: error,
```

Comment:
```JavaScript
  getErrorMessage: function (fieldId) {
    var state = this.state;
    var errorIndex = state.errorIndices[fieldId];
    if (errorIndex != null && !Util.isArray(errorIndex)) {
      return AppFormErrorMessages.getFieldMessage(fieldId, errorIndex);
    }
    if (state.responseErrorMessages[fieldId] != null) {
      return state.responseErrorMessages[fieldId];
    }
    return null;
  },
```
The `null` results sometimes from this function in AppConfigEditFormComponent.jsx.
I've noticed that not checking for `error != null` sometimes resulted in a disabled submit button mistakenly.